### PR TITLE
Refactor test result parsing in NetfilterStater class

### LINKED ISSUE #262: stats/kernel-selftests: resolve rubocop issues of NetfilterStater class
### Reproduce
```
$ rubocop --only Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity stats/kernel-selftests
stats/kernel-selftests:1107:3: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for stat is too high. [13/7]
  def stat(line, stats) ...
  ^^^^^^^^^^^^^^^^^^^^^
stats/kernel-selftests:1107:3: C: Metrics/PerceivedComplexity: Perceived complexity for stat is too high. [10/8]
  def stat(line, stats) ...
  ^^^^^^^^^^^^^^^^^^^^^
```
### Reference
https://github.com/intel/lkp-tests/wiki/Coding-Standard#rubocop

### Acceptance Test
```
$ rubocop stats/kernel-selftests # there's no issue of NetfilterStater class
$ rake spec spec=stats # pass
```

### PR DESCRIPTION
**Pull request by fongjiantan**
Fixes #262

### CODE REVIEW

File: stats/kernel-selftests [hunk 1]
```diff
@@ -1112,37 +1112,50 @@ class NetfilterStater < Stater
       @test_case = $1 if @test_script == 'nft_concat_range.sh'
     when /^#   (.+)\[( OK|FAIL|SKIP)/
       #   Add two elements, flush, re-add                               [ OK ]
-      @subtest_case = $1.strip if @test_script == 'nft_concat_range.sh'
-      stats.add "#{@test_prefix}.#{@test_case}.#{$1}", $2 if @test_script == 'nft_concat_range.sh'
+      test_case = $1
+      result = $2
+      parse_subtest_result(test_case, result, stats)
```
Comment (rli9) [inline stats/kernel-selftests:1117]: thanks, it's ok here to just call arse_subtest_result($1, $2, stats) without earlier 2 assignments

File: stats/kernel-selftests [hunk 2]
```diff
@@ -1112,37 +1112,50 @@ class NetfilterStater < Stater
...
+
+  def parse_subtest_result(test_case, result, stats)
+    return unless @test_script == 'nft_concat_range.sh'
+
+    @subtest_case = test_case.strip
+    stats.add "#{@test_prefix}.#{@test_case}.#{test_case}", result
+  end
+
+  def parse_test_result(test_case, result, stats)
+    if test_case =~ /(ns\d)-\S+ (connection on port.*)/
```
Comment (rli9) [inline stats/kernel-selftests:1138]: prefer to match different line outside of this function, so in the function, better to only wrap the logic of "when /^# (PASS|FAIL|SKIP|OK): (.*)/" part.

Also two when clauses are slightly different, one can match OK, one can't. Thus combine it to match (PASS|FAIL|SKIP|OK): (.*) may bring some confusion.

**Comment by fongjiantan** [comment]
@rli9 I made quite a few changes based on your feedback. The cyclomatic complexity metric is quite sensitive haha. But I have tested the code using `rake spec spec=stats` and all tests are passing

**Comment by rli9** [comment]
thanks, i will take the patch. By going through the change and your comment "cyclomatic complexity metric is quite sensitive", I think maybe we need loose the criteria in some cases, so the fix can focus on most complex code to increase the readability.

**Pull request merged**